src/go/mongolib/util/util.go

```diff
@@ -119,6 +122,9 @@ func GetHostnames(ctx context.Context, client *mongo.Client) ([]string, error) {
 	var shardsMap proto.ShardsMap
 	smRes := client.Database("admin").RunCommand(ctx, primitive.M{"getShardMap": 1})
 	if smRes.Err() != nil {
+		if e, ok := smRes.Err().(mongo.CommandError); ok && e.Code == shardingNotEnabledError {
+			return nil, nil // standalone instance
```

Comment:
I would suggest to return error like ErrShardingNotEnabled and handle it where we call this method.